guides/compatibility-mode.md

```diff
@@ -0,0 +1,32 @@
+# Compatibility Mode
+Compatibility mode allows `@angular/material` to work with [Angular Material for AngularJS](https://material.angularjs.org/latest/).
```

Comment:
I would open this with

```
Compatibility mode for `@angular/material` allows the components to exist side-by-side with AngularJS Material components without any CSS collisions.
```

As for the structure of the doc, I think the it should be something like
```md
## What is compatibility mode?
Compatibility mode for `@angular/material` allows the components to exist side-by-side with AngularJS Material components without any CSS collisions.

## What does compatibility mode do?
When enabled, compatibility mode enforces that all template APIs use the prefix `mat-` instead of `md-`. This will prevent any CSS from AngularJS Material from affecting the components in `@angular/material`.

## How is compatibility mode enabled?
Simply import `NoConflictStyleCompatibilityMode` into your application's root `NgModule`. 
```